CMakeLists.txt

```diff
@@ -77,7 +77,7 @@ PROJECT (CLICKHOUSE-CLIENT)
     ENDIF ()
 
     IF (WITH_SYSTEM_CITYHASH)
-        FIND_PACKAGE(cityhash REQUIRED)
```

Comment:
Please do not use find_library, you'll break compatibility with package managers. Look at https://github.com/ClickHouse/clickhouse-cpp/pull/168#issuecomment-1367840372

Reply:
@Jihadist could you please elaborate on "compatibility with package managers"? cityhash does not ship any CMake config file (not even a pkg-config one), so it doesn't seem impossible to find `cityhash` with `find_package`.

Disclaimer: I am a maintainer of the [Homebrew](https://brew.sh/) package manager. In https://github.com/Homebrew/homebrew-core/pull/128575, we could not find `cityhash` without fixing this. (Though now we are going to use the vendored `cityhash` due to hash compatibility concerns as mentioned in https://github.com/ClickHouse/clickhouse-cpp/pull/301#issuecomment-1520592157. But that's a different point.)

Reply:
@ZhongRuoyu  vcpkg and conan provide cityhash as _cityhash_ cmake target. I mentioned comment especially because author provide an example of conanfile for clickhouse-cpp (it was not actually merged but it another question)

Reply:
Well, I'd argue that it's not the general case. As I mentioned earlier, cityhash does *not* provide any config files, so it is the user's (project's) responsibility to find out where the cityhash library is. To be clearer: package managers can provide their own CMake target, but I don't think projects should rely on that, because it is not the general case. For instance, none of Homebrew, [FreeBSD](https://github.com/freebsd/freebsd-ports/blob/main/devel/cityhash/pkg-plist), or [Raspbian](http://archive.raspbian.org/raspbian/pool/main/c/cityhash/) provide anything more than the headers and the library.

Nevertheless, since it wouldn't hurt anything, let me unbreak that by attempting `find_package` first.

Reply:
> As I mentioned earlier, cityhash does not provide any config files, so it is the user's (project's) responsibility to find out where the cityhash library is. To be clearer: package managers can provide their own CMake target, but I don't think projects should rely on that, because it is not the general case. For instance, none of Homebrew, [FreeBSD](https://github.com/freebsd/freebsd-ports/blob/main/devel/cityhash/pkg-plist), or [Raspbian](http://archive.raspbian.org/raspbian/pool/main/c/cityhash/) provide anything more than the headers and the library.

Yep, you are right here. But it's not an argument to use `find_library `in root cmakelists. `find_library `should be used inside custom cmake find*.cmake script and this script may import target. So you do not need to remove find_package, and it will work with your homebrew or bsd package.

Reply:
Your current solution is not bad too. Maybe it ok

Reply:
Thanks -- @Enmk also pointed this out. This was out of my knowledge, but indeed it looks more compatible with the current CMake configs. I've pushed a `Findcityhash.cmake` script and that should help to find cityhash.